pkg/app/server/grpcapi/web_api.go

```diff
@@ -1535,6 +1535,10 @@ func (a *WebAPI) GetInsightData(ctx context.Context, req *webservice.GetInsightD
 		return nil, err
 	}
 
+	if !config.FeatureFlagEnabled(config.FeatureFlagInsights) {
```

Comment:
nits, probably move it before extract claims ctx is better to avoid unnecessary handling

Reply:
Yes, it should be under the auth protection.